neuralmonkey/runners/base_runner.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any, Dict, Tuple, List, NamedTuple, Union
+import numpy as np
+import tensorflow as tf
+
+# tests: pylint, mypy
+
+# pylint: disable=invalid-name
+FeedDict = Dict[tf.Tensor, Union[int, float, np.ndarray]]
+NextExecute = Tuple[List[Any], List[tf.Tensor], FeedDict]
+RunResult = Union[float, np.ndarray, tf.Summary]
+ExecutionResult = NamedTuple('ExecutionResult',
+                             [('outputs', List[Any]),
+                              ('losses', List[float]),
+                              ('scalar_summaries', tf.Summary),
+                              ('histogram_summaries', tf.Summary),
+                              ('image_summaries', tf.Summary)])
+
+
+class Executable(object):
+
+    def next_to_execute(self) -> NextExecute:
+        raise NotImplementedError()
+
+    def collect_results(self, results: List[List[RunResult]]) -> None:
+        raise NotImplementedError()
+
+
+def collect_encoders(coder):
+    """Collect recusively all encoders and decoders."""
+    if hasattr(coder, 'encoders'):
+        return set([coder]).union(*(collect_encoders(enc)
+                                    for enc in coder.encoders))
+    else:
+        return set([coder])
+
+
+class BaseRunner(object):
+    def __init__(self, output_series: str, decoder) -> None:
```

Comment:
Aspoň Any, nebo tak něco.. (?) Souvisí to s mojí otázkou nahoře, co je víc pythonický..